scripts/httpd/upload.cgi

```diff
@@ -160,6 +160,45 @@ def emit_fedmsg(config, name, checksum, filename, username, msgpath):
         except Exception as e:
             sys.stderr.write("Error with fedmsg", str(e))
 
+    # Emit a fedmsg-messaging message
+    if config.getboolean('upload', 'fedora-messaging', fallback=True):
+        try:
+            import fedora_messaging.api
+            import fedora_messaging.config
+            import fedora_messaging.exceptions
+        except ImportError:
+            sys.stderr.write(
+                "fedora-messasing must be installed for notifications to work")
```

Comment:
You should probably return from the function here so you don't execute the rest of the code. Or better: put all this inside its own function, then you can safely return from it (because otherwise someone may add code after the fedora-messaging block and not realize there's an early return above).

Reply:
would you have a patch for this?

Reply:
Yes, give me access to your fork and I'll push it